bundles/org.dataflowanalysis.analysis/src/org/dataflowanalysis/analysis/core/AbstractVertex.java

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /**
+     * Returns the referenced model element by the abstract vertex. Multiple vertices may reference the same model element,
+     * but one vertex always references exactly one model element.
+     * @return Returns the reference element by the vertex
+     */
+    public T getReferencedElement() {
+        return referencedElement;
+    }
+
+    /**
+     * Returns a list of all data flow variables that are present for the action sequence element
+     * @return List of present data flow variables
+     */
+    public List<DataFlowVariable> getAllDataFlowVariables() {
+        return this.incomingDataFlowVariables.orElseThrow(IllegalStateException::new);
+    }
+
+    /**
+     * Returns a list of all incoming data flow variables that are present for the action sequence element
+     * @return List of present incoming data flow variables (e.g. the variables at the input pin of the DFD representation)
+     */
+    public List<DataFlowVariable> getAllIncomingDataFlowVariables() {
+        return this.incomingDataFlowVariables.orElseThrow(IllegalStateException::new);
+    }
+
+    /**
+     * Returns a list of all outgoing data flow variables that are present for the action sequence element
+     * @return List of present outgoing data flow variables (e.g. the variables at the output pin of the DFD representation)
+     */
+    public List<DataFlowVariable> getAllOutgoingDataFlowVariables() {
+        return this.outgoingDataFlowVariables.orElseThrow(IllegalStateException::new);
+    }
+
+    /**
+     * Returns a list of all present node characteristics for the action sequence element
+     * @return List of present node characteristics
+     */
+    public List<CharacteristicValue> getAllNodeCharacteristics() {
+        return this.vertexCharacteristics.orElseThrow(IllegalStateException::new);
+    }
+
+    /**
+     * Returns the list of previous elements that precede the vertex
+     * @return Returns a list of all preceding vertices of the vertex.
+     */
+    public abstract List<? extends AbstractVertex<?>> getPreviousElements();
+
+    /**
+     * Returns whether the vertex is a source (e.g. does not have a source)
+     * @return Returns true, if the vertex is a source. Otherwise, the method returns false
+     */
+    public boolean isSource() {
+        return this.getPreviousElements().isEmpty();
+    }
+
+    /**
+     * Returns a list of characteristic literals that are set for a given characteristic type in the list of all node
+     * characteristics
+     * <p>
+     * See {@link AbstractVertex#getDataFlowCharacteristicNamesWithType(String)} for a similar method for data flow
+     * variables
+     * @param name Name of the characteristic type
+     * @return Returns a list of all characteristic literals matching the characteristic type
+     */
+    public List<String> getNodeCharacteristicNamesWithName(String name) {
```

Comment:
I have problems understanding the documentation of this and the following methods, as there is no clear distinction between names and IDs

Reply:
Removed methods and replaced with better documented methods in 8f74e1c836e0a214fc8c9e3f5ae0f2214ea2032d.

We should probably discuss how we want to handle common lookups for data flow variables, data flow characteristics (e.g. characteristic values regardless of data flow variable name) and node characteristics

Reply:
Moved to #130